### PR TITLE
feat: add support for custom and auto agent size

### PR DESCRIPTION
**Pull request by hanxiaop**
Fixes https://github.com/akuity/terraform-provider-akp/issues/259

### CODE REVIEW

File: docs/resources/cluster.md [hunk 1]
```diff
@@ -152,7 +152,9 @@ Required:
 Optional:
 
 - `app_replication` (Boolean) Enables Argo CD state replication to the managed cluster that allows disconnecting the cluster from Akuity Platform without losing core Argocd features
+- `auto_agent_size_config` (Attributes) Autoscaler config for auto agent size (see [below for nested schema](#nestedatt--spec--data--auto_agent_size_config))
```
Comment (jiachengxu) [inline docs/resources/cluster.md:155]: I think we will need to add `auto` and `custom` to  https://github.com/akuity/terraform-provider-akp/blob/main/akp/data_source_akp_cluster_schema.go#L84 and
https://github.com/akuity/terraform-provider-akp/blob/main/akp/resource_akp_cluster_schema.go#L145

File: examples/resources/akp_cluster/auto_agent_size.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
+data "akp_instance" "example" {
```
Comment (jiachengxu) [inline examples/resources/akp_cluster/auto_agent_size.tf:1]: let's modify the template to include this and custom agent size example in the generated docs. We can just add them after here: https://github.com/akuity/terraform-provider-akp/blob/main/templates/resources/cluster.md.tmpl#L14-L20

**Comment by hanxiaop** [comment]
> one nitpick: I would use `memory` instead of `mem` like how UI shows it. image

@junzebao Thank you! Fixed.

File: akp/data_source_akp_cluster_schema.go [hunk 3]
```diff
@@ -201,3 +211,114 @@ func getManagedClusterConfigDataSourceAttributes() map[string]schema.Attribute {
...
+		"replica_minimum": schema.Int64Attribute{
+			Description: "Replica minimum",
+			Computed:    true,
+		},
+	}
+}
+
+func getResourcesDataSourceAttributes() map[string]schema.Attribute {
+	return map[string]schema.Attribute{
+		"mem": schema.StringAttribute{
```
Comment (jiachengxu) [inline akp/data_source_akp_cluster_schema.go:315]: based on @junzebao 's comment, this also need to be "memory“？

File: akp/resource_akp_cluster.go [hunk 4]
```diff
@@ -275,6 +276,9 @@ func buildClusterApplyRequest(ctx context.Context, diagnostics *diag.Diagnostics
 func buildClusters(ctx context.Context, diagnostics *diag.Diagnostics, cluster *types.Cluster) []*structpb.Struct {
 	var cs []*structpb.Struct
 	apiCluster := cluster.ToClusterAPIModel(ctx, diagnostics)
+	if diagnostics.HasError() {
```
Comment (jiachengxu) [inline akp/resource_akp_cluster.go:279]: what is the reason that we ignore the error here?

File: akp/resource_akp_cluster_schema.go [hunk 5]
```diff
@@ -346,3 +358,150 @@ func getManagedClusterConfigAttributes() map[string]schema.Attribute {
...
+	return map[string]schema.Attribute{
+		"cpu": schema.StringAttribute{
+			Description: "CPU",
+			Optional:    true,
+			Computed:    true,
+			PlanModifiers: []planmodifier.String{
+				stringplanmodifier.UseStateForUnknown(),
+			},
+		},
+		"mem": schema.StringAttribute{
```
Comment (jiachengxu) [inline akp/resource_akp_cluster_schema.go:498]: "memory" here?

File: examples/resources/akp_cluster/auto_agent_size.tf [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+          replica_maximum = 3,
+          # minimum number of replicas should be set to 1
+          replica_minimum = 1,
+          resource_maximum = {
+            cpu = "3"
+            mem = "2.00Gi"
+          },
+          resource_minimum = {
+            cpu = "250m",
+            mem = "256Mi"
```
Comment (jiachengxu) [inline examples/resources/akp_cluster/auto_agent_size.tf:43]: memory?

File: akp/data_source_akp_cluster_schema.go (hunk 3 above)
Comment (hanxiaop) [reply akp/data_source_akp_cluster_schema.go]: Revised. Was only revised custom agent size config, and now I've also revised auto agent size config. Thank you!

File: akp/resource_akp_cluster.go (hunk 4 above)
Comment (hanxiaop) [reply akp/resource_akp_cluster.go:279]: @jiachengxu This doesn't ignore error but rather exits early when having errors, for example input validation error. And then the diagnostic errors will be reported to users.

**Review by junzebao — APPROVED** [review]

**Review by junzebao — APPROVED** [review]

**Pull request merged**